libyul/AsmParser.cpp

```diff
@@ -222,6 +231,37 @@ optional<pair<string_view, SourceLocation>> Parser::parseSrcComment(
 	return {{tail, SourceLocation{}}};
 }
 
+optional<pair<string_view, optional<int>>> Parser::parseASTIDComment(
+	string_view _arguments,
+	langutil::SourceLocation const& _commentLocation
+)
+{
+	static regex const argRegex = regex(
+		R"~~(^(\d+))~~",
```

Comment:
For consistency with `@src` I think this should require whitespace after a number:
```suggestion
		R"~~(^(\d+)(?:\s+|$))~~",
```

Also, we should have a test for this case.

Reply:
Also, we should have a test case with an ID larger than `int64_t`.